packages/pds/src/actor-store/index.ts

```diff
@@ -83,44 +53,46 @@ export class ActorStore {
   }
 
   async keypair(did: string): Promise<Keypair> {
-    const got = await this.keyCache.fetch(did)
-    if (!got) {
-      throw new InvalidRequestError('Keypair not found', 'NotFound')
-    }
-    return got
+    const { keyLocation } = await this.getLocation(did)
+    const privKey = await fs.readFile(keyLocation)
+    return crypto.Secp256k1Keypair.import(privKey)
   }
 
-  async db(did: string): Promise<ActorDb> {
-    const got = await this.dbCache.fetch(did)
-    if (!got) {
+  async openDb(did: string): Promise<ActorDb> {
+    const { dbLocation } = await this.getLocation(did)
+    const exists = await fileExists(dbLocation)
+    if (!exists) {
       throw new InvalidRequestError('Repo not found', 'NotFound')
     }
-    return got
-  }
 
-  async reader(did: string) {
-    const [db, keypair] = await Promise.all([this.db(did), this.keypair(did)])
-    return createActorReader(did, db, keypair, this.resources)
+    return getDb(dbLocation, this.cfg.disableWalAutoCheckpoint)
   }
 
   async read<T>(did: string, fn: ActorStoreReadFn<T>) {
-    const reader = await this.reader(did)
-    return fn(reader)
+    const keypair = await this.keypair(did)
+    const db = await this.openDb(did)
+    try {
+      const reader = createActorReader(did, db, keypair, this.resources)
+      return await fn(reader)
+    } finally {
+      await db.close()
```

Comment:
I'm not sure that we need to `await` the close here, it may be fine to let node just manage those in the background. 
 Though in that case especially, we'd want to `catch()` any log errors that may result during the close to avoid unhandled rejections.

Reply:
Ended up making `close` on the database class synchronous & moving the `catch()` logic to that method

Thinking is that it should protect against more footguns in catching errors while closing dbs, such as in the callbacks on streams (elsewhere in this PR)